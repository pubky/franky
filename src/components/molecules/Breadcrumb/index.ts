export * from './Breadcrumb';
export * from './Breadcrumb.types';
export * from './Breadcrumb.examples';
