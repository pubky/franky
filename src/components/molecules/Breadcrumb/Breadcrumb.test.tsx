import React from 'react';
import { describe, it, expect, vi } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import { Breadcrumb, BreadcrumbItem, BreadcrumbSeparator, BreadcrumbEllipsis, BreadcrumbPage } from './Breadcrumb';

describe('Breadcrumb', () => {
  describe('Breadcrumb Container', () => {
    it('renders with nav element', () => {
      render(
        <Breadcrumb>
          <BreadcrumbItem>Home</BreadcrumbItem>
        </Breadcrumb>,
      );

      const nav = screen.getByRole('navigation', { name: 'breadcrumb' });
      expect(nav).toBeInTheDocument();
    });

    it('renders with ordered list', () => {
      const { container } = render(
        <Breadcrumb>
          <BreadcrumbItem>Home</BreadcrumbItem>
        </Breadcrumb>,
      );

      const ol = container.querySelector('ol');
      expect(ol).toBeInTheDocument();
    });
  });

  describe('BreadcrumbItem', () => {
    it('renders as list item', () => {
      render(
        <Breadcrumb>
          <BreadcrumbItem>Home</BreadcrumbItem>
        </Breadcrumb>,
      );

      const listItem = screen.getByRole('listitem');
      expect(listItem).toBeInTheDocument();
    });

    it('renders as link when href is provided', () => {
      render(
        <Breadcrumb>
          <BreadcrumbItem href="/home">Home</BreadcrumbItem>
        </Breadcrumb>,
      );

      const link = screen.getByRole('link');
      expect(link).toHaveAttribute('href', '/home');
    });

    it('renders as button when href is not provided', () => {
      render(
        <Breadcrumb>
          <BreadcrumbItem>Home</BreadcrumbItem>
        </Breadcrumb>,
      );

      const button = screen.getByRole('button');
      expect(button).toBeInTheDocument();
    });

    it('handles click events', () => {
      const handleClick = vi.fn();
      render(
        <Breadcrumb>
          <BreadcrumbItem onClick={handleClick}>Click me</BreadcrumbItem>
        </Breadcrumb>,
      );

      const item = screen.getByRole('listitem');
      fireEvent.click(item);
      expect(handleClick).toHaveBeenCalledTimes(1);
    });
  });

  describe('Accessibility', () => {
    it('maintains keyboard navigation', () => {
      const handleClick = vi.fn();
      render(
        <Breadcrumb>
          <BreadcrumbItem onClick={handleClick}>Clickable</BreadcrumbItem>
        </Breadcrumb>,
      );

      const button = screen.getByRole('button');
      expect(button).toBeInTheDocument();

      // Buttons are keyboard accessible by default
      fireEvent.click(button);
      expect(handleClick).toHaveBeenCalled();
    });
  });
});

describe('Breadcrumb - Snapshots', () => {
  describe('Breadcrumb Container', () => {
    it('matches snapshot with default props', () => {
      const { container } = render(
        <Breadcrumb>
          <BreadcrumbItem>Home</BreadcrumbItem>
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot with small size', () => {
      const { container } = render(
        <Breadcrumb size="sm">
          <BreadcrumbItem>Home</BreadcrumbItem>
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot with medium size', () => {
      const { container } = render(
        <Breadcrumb size="md">
          <BreadcrumbItem>Home</BreadcrumbItem>
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot with custom className', () => {
      const { container } = render(
        <Breadcrumb className="custom-breadcrumb">
          <BreadcrumbItem>Home</BreadcrumbItem>
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('BreadcrumbItem', () => {
    it('matches snapshot with link variant', () => {
      const { container } = render(
        <Breadcrumb>
          <BreadcrumbItem variant="link">Home</BreadcrumbItem>
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot with current variant', () => {
      const { container } = render(
        <Breadcrumb>
          <BreadcrumbItem variant="current">Current</BreadcrumbItem>
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot with dropdown variant', () => {
      const { container } = render(
        <Breadcrumb>
          <BreadcrumbItem dropdown>Menu</BreadcrumbItem>
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot with href prop', () => {
      const { container } = render(
        <Breadcrumb>
          <BreadcrumbItem href="/home">Home</BreadcrumbItem>
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot with custom className', () => {
      const { container } = render(
        <Breadcrumb>
          <BreadcrumbItem className="custom-item">Home</BreadcrumbItem>
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('BreadcrumbSeparator', () => {
    it('matches snapshot with default icon', () => {
      const { container } = render(
        <Breadcrumb>
          <BreadcrumbItem>Home</BreadcrumbItem>
          <BreadcrumbSeparator />
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot with small size', () => {
      const { container } = render(
        <Breadcrumb>
          <BreadcrumbItem>Home</BreadcrumbItem>
          <BreadcrumbSeparator size="sm" />
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot with medium size', () => {
      const { container } = render(
        <Breadcrumb>
          <BreadcrumbItem>Home</BreadcrumbItem>
          <BreadcrumbSeparator size="md" />
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot with custom icon', () => {
      const { container } = render(
        <Breadcrumb>
          <BreadcrumbItem>Home</BreadcrumbItem>
          <BreadcrumbSeparator icon={<span data-testid="custom-separator">/</span>} />
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot with custom className', () => {
      const { container } = render(
        <Breadcrumb>
          <BreadcrumbItem>Home</BreadcrumbItem>
          <BreadcrumbSeparator className="custom-separator" />
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('BreadcrumbEllipsis', () => {
    it('matches snapshot with default props', () => {
      const { container } = render(
        <Breadcrumb>
          <BreadcrumbEllipsis />
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot with custom className', () => {
      const { container } = render(
        <Breadcrumb>
          <BreadcrumbEllipsis className="custom-ellipsis" />
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('BreadcrumbPage', () => {
    it('matches snapshot with default props', () => {
      const { container } = render(
        <Breadcrumb>
          <BreadcrumbPage>Current Page</BreadcrumbPage>
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot with custom className', () => {
      const { container } = render(
        <Breadcrumb>
          <BreadcrumbPage className="custom-page">Current Page</BreadcrumbPage>
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Complete Breadcrumb Examples', () => {
    it('matches snapshot for simple breadcrumb', () => {
      const { container } = render(
        <Breadcrumb>
          <BreadcrumbItem href="/">Home</BreadcrumbItem>
          <BreadcrumbSeparator />
          <BreadcrumbPage>Current</BreadcrumbPage>
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot for complex breadcrumb', () => {
      const { container } = render(
        <Breadcrumb>
          <BreadcrumbItem href="/">Home</BreadcrumbItem>
          <BreadcrumbSeparator />
          <BreadcrumbEllipsis />
          <BreadcrumbSeparator />
          <BreadcrumbItem dropdown>Category</BreadcrumbItem>
          <BreadcrumbSeparator />
          <BreadcrumbItem href="/products">Products</BreadcrumbItem>
          <BreadcrumbSeparator />
          <BreadcrumbPage>Product Details</BreadcrumbPage>
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot for small size breadcrumb', () => {
      const { container } = render(
        <Breadcrumb size="sm">
          <BreadcrumbItem href="/">Home</BreadcrumbItem>
          <BreadcrumbSeparator size="sm" />
          <BreadcrumbItem href="/category">Category</BreadcrumbItem>
          <BreadcrumbSeparator size="sm" />
          <BreadcrumbPage>Current</BreadcrumbPage>
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot with custom styling', () => {
      const { container } = render(
        <Breadcrumb className="custom-breadcrumb">
          <BreadcrumbItem href="/" className="custom-link">
            Home
          </BreadcrumbItem>
          <BreadcrumbSeparator className="custom-sep" icon={<span>/</span>} />
          <BreadcrumbItem variant="current" className="custom-current">
            Current
          </BreadcrumbItem>
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
