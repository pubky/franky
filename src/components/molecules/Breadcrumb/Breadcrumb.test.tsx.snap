// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Breadcrumb - Snapshots > Breadcrumb Container > matches snapshot with custom className 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5 custom-breadcrumb"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer"
    >
      <button
        class="flex items-center gap-1"
        type="button"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Home
        </span>
      </button>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > Breadcrumb Container > matches snapshot with default props 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer"
    >
      <button
        class="flex items-center gap-1"
        type="button"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Home
        </span>
      </button>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > Breadcrumb Container > matches snapshot with medium size 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer"
    >
      <button
        class="flex items-center gap-1"
        type="button"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Home
        </span>
      </button>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > Breadcrumb Container > matches snapshot with small size 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-1.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer"
    >
      <button
        class="flex items-center gap-1"
        type="button"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Home
        </span>
      </button>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > BreadcrumbEllipsis > matches snapshot with custom className 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <span
      aria-hidden="true"
      class="flex h-9 w-9 items-center justify-center custom-ellipsis"
      role="presentation"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-ellipsis h-4 w-4"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="1"
        />
        <circle
          cx="19"
          cy="12"
          r="1"
        />
        <circle
          cx="5"
          cy="12"
          r="1"
        />
      </svg>
      <span
        class="sr-only"
      >
        More
      </span>
    </span>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > BreadcrumbEllipsis > matches snapshot with default props 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <span
      aria-hidden="true"
      class="flex h-9 w-9 items-center justify-center"
      role="presentation"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-ellipsis h-4 w-4"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="1"
        />
        <circle
          cx="19"
          cy="12"
          r="1"
        />
        <circle
          cx="5"
          cy="12"
          r="1"
        />
      </svg>
      <span
        class="sr-only"
      >
        More
      </span>
    </span>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > BreadcrumbItem > matches snapshot with current variant 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <li
      class="flex items-center justify-center gap-2.5 text-foreground"
    >
      <button
        class="flex items-center gap-1"
        type="button"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Current
        </span>
      </button>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > BreadcrumbItem > matches snapshot with custom className 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer custom-item"
    >
      <button
        class="flex items-center gap-1"
        type="button"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Home
        </span>
      </button>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > BreadcrumbItem > matches snapshot with dropdown variant 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer rounded overflow-hidden"
    >
      <button
        class="flex items-center gap-1"
        type="button"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Menu
        </span>
        <svg
          aria-hidden="true"
          class="lucide lucide-chevron-down w-[15px] h-[15px]"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m6 9 6 6 6-6"
          />
        </svg>
      </button>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > BreadcrumbItem > matches snapshot with href prop 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer"
    >
      <a
        class="flex items-center gap-1"
        href="/home"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Home
        </span>
      </a>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > BreadcrumbItem > matches snapshot with link variant 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer"
    >
      <button
        class="flex items-center gap-1"
        type="button"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Home
        </span>
      </button>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > BreadcrumbPage > matches snapshot with custom className 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <span
      aria-current="page"
      aria-disabled="true"
      class="font-normal text-foreground custom-page"
      role="link"
    >
      Current Page
    </span>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > BreadcrumbPage > matches snapshot with default props 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <span
      aria-current="page"
      aria-disabled="true"
      class="font-normal text-foreground"
      role="link"
    >
      Current Page
    </span>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > BreadcrumbSeparator > matches snapshot with custom className 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer"
    >
      <button
        class="flex items-center gap-1"
        type="button"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Home
        </span>
      </button>
    </li>
    <li
      aria-hidden="true"
      class="text-muted-foreground shrink-0 w-4 h-4 custom-separator"
      role="presentation"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-right w-full h-full"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m9 18 6-6-6-6"
        />
      </svg>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > BreadcrumbSeparator > matches snapshot with custom icon 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer"
    >
      <button
        class="flex items-center gap-1"
        type="button"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Home
        </span>
      </button>
    </li>
    <li
      aria-hidden="true"
      class="text-muted-foreground shrink-0 w-4 h-4"
      role="presentation"
    >
      <span
        data-testid="custom-separator"
      >
        /
      </span>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > BreadcrumbSeparator > matches snapshot with default icon 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer"
    >
      <button
        class="flex items-center gap-1"
        type="button"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Home
        </span>
      </button>
    </li>
    <li
      aria-hidden="true"
      class="text-muted-foreground shrink-0 w-4 h-4"
      role="presentation"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-right w-full h-full"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m9 18 6-6-6-6"
        />
      </svg>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > BreadcrumbSeparator > matches snapshot with medium size 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer"
    >
      <button
        class="flex items-center gap-1"
        type="button"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Home
        </span>
      </button>
    </li>
    <li
      aria-hidden="true"
      class="text-muted-foreground shrink-0 w-4 h-4"
      role="presentation"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-right w-full h-full"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m9 18 6-6-6-6"
        />
      </svg>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > BreadcrumbSeparator > matches snapshot with small size 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer"
    >
      <button
        class="flex items-center gap-1"
        type="button"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Home
        </span>
      </button>
    </li>
    <li
      aria-hidden="true"
      class="text-muted-foreground shrink-0 w-4 h-4"
      role="presentation"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-right w-full h-full"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m9 18 6-6-6-6"
        />
      </svg>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > Complete Breadcrumb Examples > matches snapshot for complex breadcrumb 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer"
    >
      <a
        class="flex items-center gap-1"
        href="/"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Home
        </span>
      </a>
    </li>
    <li
      aria-hidden="true"
      class="text-muted-foreground shrink-0 w-4 h-4"
      role="presentation"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-right w-full h-full"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m9 18 6-6-6-6"
        />
      </svg>
    </li>
    <span
      aria-hidden="true"
      class="flex h-9 w-9 items-center justify-center"
      role="presentation"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-ellipsis h-4 w-4"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="1"
        />
        <circle
          cx="19"
          cy="12"
          r="1"
        />
        <circle
          cx="5"
          cy="12"
          r="1"
        />
      </svg>
      <span
        class="sr-only"
      >
        More
      </span>
    </span>
    <li
      aria-hidden="true"
      class="text-muted-foreground shrink-0 w-4 h-4"
      role="presentation"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-right w-full h-full"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m9 18 6-6-6-6"
        />
      </svg>
    </li>
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer rounded overflow-hidden"
    >
      <button
        class="flex items-center gap-1"
        type="button"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Category
        </span>
        <svg
          aria-hidden="true"
          class="lucide lucide-chevron-down w-[15px] h-[15px]"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m6 9 6 6 6-6"
          />
        </svg>
      </button>
    </li>
    <li
      aria-hidden="true"
      class="text-muted-foreground shrink-0 w-4 h-4"
      role="presentation"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-right w-full h-full"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m9 18 6-6-6-6"
        />
      </svg>
    </li>
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer"
    >
      <a
        class="flex items-center gap-1"
        href="/products"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Products
        </span>
      </a>
    </li>
    <li
      aria-hidden="true"
      class="text-muted-foreground shrink-0 w-4 h-4"
      role="presentation"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-right w-full h-full"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m9 18 6-6-6-6"
        />
      </svg>
    </li>
    <span
      aria-current="page"
      aria-disabled="true"
      class="font-normal text-foreground"
      role="link"
    >
      Product Details
    </span>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > Complete Breadcrumb Examples > matches snapshot for simple breadcrumb 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer"
    >
      <a
        class="flex items-center gap-1"
        href="/"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Home
        </span>
      </a>
    </li>
    <li
      aria-hidden="true"
      class="text-muted-foreground shrink-0 w-4 h-4"
      role="presentation"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-right w-full h-full"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m9 18 6-6-6-6"
        />
      </svg>
    </li>
    <span
      aria-current="page"
      aria-disabled="true"
      class="font-normal text-foreground"
      role="link"
    >
      Current
    </span>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > Complete Breadcrumb Examples > matches snapshot for small size breadcrumb 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-1.5"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer"
    >
      <a
        class="flex items-center gap-1"
        href="/"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Home
        </span>
      </a>
    </li>
    <li
      aria-hidden="true"
      class="text-muted-foreground shrink-0 w-4 h-4"
      role="presentation"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-right w-full h-full"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m9 18 6-6-6-6"
        />
      </svg>
    </li>
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer"
    >
      <a
        class="flex items-center gap-1"
        href="/category"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Category
        </span>
      </a>
    </li>
    <li
      aria-hidden="true"
      class="text-muted-foreground shrink-0 w-4 h-4"
      role="presentation"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-right w-full h-full"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m9 18 6-6-6-6"
        />
      </svg>
    </li>
    <span
      aria-current="page"
      aria-disabled="true"
      class="font-normal text-foreground"
      role="link"
    >
      Current
    </span>
  </ol>
</nav>
`;

exports[`Breadcrumb - Snapshots > Complete Breadcrumb Examples > matches snapshot with custom styling 1`] = `
<nav
  aria-label="breadcrumb"
  class="flex items-center flex-wrap gap-2.5 custom-breadcrumb"
>
  <ol
    class="flex items-center flex-wrap gap-inherit"
  >
    <li
      class="flex items-center justify-center gap-2.5 text-muted-foreground hover:text-foreground transition-colors cursor-pointer custom-link"
    >
      <a
        class="flex items-center gap-1"
        href="/"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Home
        </span>
      </a>
    </li>
    <li
      aria-hidden="true"
      class="text-muted-foreground shrink-0 w-4 h-4 custom-sep"
      role="presentation"
    >
      <span>
        /
      </span>
    </li>
    <li
      class="flex items-center justify-center gap-2.5 text-foreground custom-current"
    >
      <button
        class="flex items-center gap-1"
        type="button"
      >
        <span
          class="font-bold text-sm leading-5 font-sans"
        >
          Current
        </span>
      </button>
    </li>
  </ol>
</nav>
`;
